### PR TITLE
Introduce signal handlers.

### PR DESCRIPTION
**Pull request by chenel**
Move signal handling out of NOvA CAFAna here.  Also, introduce new functionality in handling `SIGINT` and `SIGTERM`:

* the first time the signal is received, set a flag that clients (e.g. `IFitter`, `SpectrumLoader`) within CAFAna can check to break out of their loops.
* the second time, exit immediately.

NOTE that this _**REPLACES**_ ROOT's default handling of SIGINT/SIGTERM.  This is probably ok because ROOT's default signal handling sucks, but caveat emptor.

### CODE REVIEW

File: CAFAna/Core/SignalHandlers.cxx [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// This file sets things up (when libCAFAnaCore is loaded) so that we invoke
```
Comment (cjbackhouse) [inline CAFAna/Core/SignalHandlers.cxx:1]: I think this comment should now reference libCAFAnaCoreExt?

**Review by cjbackhouse — CHANGES_REQUESTED** [review]
Minor nits

File: CAFAna/Core/SignalHandlers.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <atomic>
```
[the remark this answers is not in the document]
Comment (chenel) [reply CAFAna/Core/SignalHandlers.h]: Ah, yeah, this was leftover from the first implementation.
Fixed in acd9101ce85fb990af0194989a1bdbfa7439f28d

File: CAFAna/Core/SignalHandlers.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace ana
+{
+  /// Loops or long-running CAFAna sub-processes (e.g. Loaders::Go(), IFitter::Fit(), ...)
+  /// can check this flag periodically to see if they are requested to quit.
+  /// CAFAna's signal handlers for SIGTERM and SIGINT set it
+  bool CAFAnaQuitRequested();
+}
```
[the remark this answers is not in the document]
Comment (chenel) [reply CAFAna/Core/SignalHandlers.h]: if you insist... acd9101ce85fb990af0194989a1bdbfa7439f28d

File: CAFAna/Core/SignalHandlers.cxx (hunk 1 above)
Comment (chenel) [reply CAFAna/Core/SignalHandlers.cxx]: ah, yes.
1dd934e97a14d815edd17e8184f06534a40fc8f4

**Review by cjbackhouse — APPROVED** [review]

**Pull request merged**